packages/backend-function/src/factory.ts

```diff
@@ -23,6 +23,7 @@ import {
   SsmEnvironmentEntry,
   StackProvider,
 } from '@aws-amplify/plugin-types';
+import type { AmplifyData } from '@aws-amplify/data-construct';
```

Comment:
We should not take dependency on `@aws-amplify/data-construct` here.

We should introduce type for `DataResources` like this one https://github.com/aws-amplify/amplify-backend/blob/main/packages/plugin-types/src/auth_resources.ts .

And then use resource provider interface like here https://github.com/aws-amplify/amplify-backend/blob/10ef35d2fe00025fefd12603bfb2e3dce696be9a/packages/backend-data/src/factory.ts#L100-L102 .

I.e. the compile time dependencies should be narrowed down types, not classes.

Reply:
Removed / updated